# MD026 - Trailing punctuation in heading

**Tags:** `headings`

**Aliases:** `no-trailing-punctuation`

**Fixable:** Some violations can be fixed by tooling

## Description

This rule is triggered on any heading that has one of the specified normal or
full-width punctuation characters as the last character in the line:

```markdown
# This is a heading.
```

To fix this, remove the trailing punctuation:

```markdown
# This is a heading
```

## Parameters

- `punctuation`: Punctuation characters (`string`, default `.,;:!。，；：！`)

## Configuration

The `punctuation` parameter can be used to specify what characters count
as punctuation at the end of a heading. For example, you can change it to
`".,;:"` to allow headings that end with an exclamation point. `?` is
allowed by default because of how common it is in headings of FAQ-style
documents. Setting the `punctuation` parameter to `""` allows all characters -
and is equivalent to disabling the rule.

Example configuration:

```toml
[linters.severity]
no-trailing-punctuation = "err"

[linters.settings.no-trailing-punctuation]
punctuation = ".,;:"  # Custom punctuation (excludes ! and full-width chars)
```

## Exceptions

The trailing semicolon of [HTML entity references][html-entity-references]
like `&copy;`, `&#169;`, and `&#x000A9;` is ignored by this rule.

GitHub emoji codes (gemoji) like `:smile:` and `:heart:` are also ignored
by this rule.

## Rationale

Headings are not meant to be full sentences. More information:
[Punctuation at the end of headers][end-punctuation].

## Examples

### Valid

```markdown
# This is a good heading

## Another good heading

### FAQ: What is this?

#### How do I use this?

##### Copyright &copy; 2023

###### Happy face :smile:
```

### Invalid

```markdown
# This heading has a period.

## This heading has an exclamation!

### This heading has a comma,

#### This heading has a semicolon;

##### This heading has a colon:

###### Multiple periods...
```

### Setext headings

Both ATX and setext style headings are checked:

```markdown
This is invalid.
================

This is also invalid!
---------------------
```

[end-punctuation]: https://cirosantilli.com/markdown-style-guide#punctuation-at-the-end-of-headers
[html-entity-references]: https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references